src/markerIcons.ts

```diff
@@ -14,39 +14,102 @@ export function getIconFromRules(
     // @ts-ignore
     const backupL = L;
     try {
-        // @ts-ignore
-        L = localL;
-        // The behavior of Leaflet Extra Markers is to render Fonr Awesome with Web Fonts & CSS, which has
-        // proven too slow for displaying hundreds of markers.
-        // This overrides the HTML generated by Extra Market to use the SVG Symbols alternative, which
-        // seems much faster.
-        // See here for more details: https://fontawesome.com/v5/docs/web/advanced/svg-symbols
-        const iconId = iconCache.getIconIdAndCache(iconSpec);
-        iconSpec.innerHTML = `<svg class="map-view-icon"><use xlink:href="#${iconId}"></use></svg>`;
-        return leaflet.ExtraMarkers.icon(iconSpec);
+        // If the shape is set to resizable-circle we will draw the circle marker itself. This makes it
+        // possible to resize the marker (make it grow or shrink) based on the number of edges (lines)
+        // connected to a given node. Otherwise, we let leaflet.ExtraMarkers handle the drawing of marker
+        // icons.
+        if (shape === 'resizable-circle') {
+            return createCircleMarker(iconSpec.markerColor, [
+                iconSpec.icon,
+                iconSpec.prefix,
+            ]);
+        } else {
+            // @ts-ignore
+            L = localL;
+            // The behavior of Leaflet Extra Markers is to render Font Awesome with Web Fonts & CSS, which has
+            // proven too slow for displaying hundreds of markers.
+            // This overrides the HTML generated by Extra Markers to use the SVG Symbols alternative, which
+            // seems much faster.
+            // See here for more details: https://fontawesome.com/v5/docs/web/advanced/svg-symbols
+            const iconId = iconCache.getIconIdAndCache(iconSpec);
+            iconSpec.innerHTML = `<svg class="map-view-icon"><use xlink:href="#${iconId}"></use></svg>`;
+            return leaflet.ExtraMarkers.icon(iconSpec);
+        }
     } finally {
         // @ts-ignore
         L = backupL;
     }
 }
 
+export function createCircleMarkerBasedOnDegree(
+    color: string = 'red',
+    iconClasses: string[] = ['fas', 'fa-person'],
+    degree: number = 0,
+    degrees: number[] = []
+): leaflet.DivIcon {
+    let size = 20;
+    let anchor = 10;
+    let step = degrees.length <= 5 ? 1 : Math.ceil(degrees.length / 5);
+    for (let i = step - 1; i < degrees.length; i += step) {
+        if (degree <= degrees[i]) {
+            // we found the correct bucket/percentile
+            break;
+        }
+        size += 4;
+        anchor += 2;
+    }
+    return createCircleMarker(color, iconClasses, size, anchor);
+}
+
+export function createCircleMarker(
+    color: string = 'red',
+    iconClasses: string[] = ['fas', 'fa-person'],
+    iconSize: number = 30,
+    iconAnchor: number = 15
+): leaflet.DivIcon {
+    let circleStyle = `
```

Comment:
The styles that don't change here as part of the icon size would do better in the styles.css file.